x/distribution/keeper/allocation.go

```diff
@@ -79,19 +79,51 @@ func (k Keeper) AllocateTokens(
 
 	// calculate fraction allocated to validators
 	communityTax := k.GetCommunityTax(ctx)
-	voteMultiplier := sdk.OneDec().Sub(proposerMultiplier).Sub(communityTax)
```

Comment:
The most important change in this PR. Please review carefully and ensure it's correct, sound, and makes sense :)

Reply:
@alexanderbez I had a question about this whole part. Is there a reason for calculating the percentage of foundation tax for each voter, instead of just removing a block amount at the end?

E.g. Something like

```go
	// calculate fraction allocated to validators
	communityTax := k.GetCommunityTax(ctx)
	foundationTax := k.GetSecretFoundationTax(ctx)

	// Define the vote multiplier as (100 - proposerMultiplier - communityTax - foundationTax)%.
	// The value along with a validator's power fraction, is used to determine the
	// validator's rewards. Note, foundationTax may be zero.
	voteMultiplier := sdk.OneDec().Sub(proposerMultiplier).Sub(communityTax).Sub(foundationTax)

	// allocate tokens proportionally to voting power minus any taxes
	for _, vote := range previousVotes {
		validator := k.stakingKeeper.ValidatorByConsAddr(ctx, vote.Validator.Address)
		powerFraction := sdk.NewDec(vote.Validator.Power).QuoTruncate(sdk.NewDec(totalPreviousPower))

		// To determine the allocation to the foundation tax, we determine the
		// validator's reward with and without the foundation tax, and take the
		// difference. Note, foundationTax may be zero. When non-zero, the invariant
		// rewardWithoutFT > reward must hold.
		reward := feesCollected.MulDecTruncate(voteMultiplier).MulDecTruncate(powerFraction)

		// allocate tokens to the validator
		k.AllocateTokensToValidator(ctx, validator, reward)

		remaining = remaining.Sub(reward)
	}

	foundationTaxSum := feesCollected.MulDecTruncate(foundationTax)
	remaining = remaining.Sub(foundationTaxSum)
...
```

Reply:
Because you want to tax each voter. The same reason we do this for the community tax.

Reply:
Hmm, am I missing something with the math? You're already taxing each voter here: 
`voteMultiplier := sdk.OneDec().Sub(proposerMultiplier).Sub(communityTax).Sub(foundationTax)`
Since they're receiving a smaller reward at:
`feesCollected.MulDecTruncate(voteMultiplier).MulDecTruncate(powerFraction)`

I suggested this because I saw the community tax works the same way

Reply:
I don't follow...You want to tax on the remaining rewards?

Reply:
No no, I mean calculate the entire tax independent of each voter's rewards, then subtract the entire tax at the end.
For example, if you know the foundation tax is 15%, can't you just do `taxAmount = 0.15 * totalRewards` instead of going through each validator and adding 15% of their rewards?

Reply:
You cannot do that as you must include the proposer multiplier. It's the same as the community tax. Maybe you can provide an alternative implementation?

Reply:
Ah, I see. Do you think something like this would work instead?

```diff
	proposerMultiplier := baseProposerReward.Add(bonusProposerReward.MulTruncate(previousFractionVotes))
	proposerReward := feesCollected.MulDecTruncate(proposerMultiplier)

	// pay previous proposer
	remaining := feesCollected
....
        remaining = remaining.Sub(proposerReward)
....

	// calculate fraction allocated to validators
	communityTax := k.GetCommunityTax(ctx)
	foundationTax := k.GetSecretFoundationTax(ctx)

+	foundationTaxSum := remaining.MulDecTruncate(foundationTax)
+       remaining = remaining.Sub(foundationTaxSum)
	
        // Define the vote multiplier as (100 - proposerMultiplier - communityTax - foundationTax)%.
	// The value along with a validator's power fraction, is used to determine the
	// validator's rewards. Note, foundationTax may be zero.
	voteMultiplier := sdk.OneDec().Sub(proposerMultiplier).Sub(communityTax).Sub(foundationTax)

	....
	
...
```

Reply:
Yes, this actually just came to my mind as well. I think it's semantically equivalent. I'll give it a shot to make sure the invariants still hold. If they do, I'll change it.

Reply:
Cool, thanks. Let me know if that ends up working:)

Reply:
Revised. Going to resolve this.